lib/services/product-manager.js

```diff
@@ -316,18 +316,28 @@ export default class ProductManager {
   /**
    * Method will take a product with variants which should be anonymized.
    * It will modify slug, key and name so the product can be created without
-   * issues with duplicate fields
+   * issues with duplicate fields.
+   * If it was anonymized before it will only replace old salt with the new one.
+   *
    * @param product productDraft
    */
   getAnonymizedProductDraft (product) {
     const salt = this._getSalt()
+    const oldAnonymizationId = product.slug[constant.PRODUCT_ANONYMIZE_SLUG_KEY]
 
     // make all languages in slug unique
     for (const lang in product.slug) // eslint-disable-line guard-for-in
-      product.slug[lang] += `-${salt}`
+      if (oldAnonymizationId)
+        product.slug[lang] = product.slug[lang].replace(oldAnonymizationId, salt)
```

Comment:
I guess in this case we will lose the original slug definition -> new anonymizatio will contain the salt only? Is there an easy way to replace the previous salt only? If not I wonder if for the sake of consistency we should save salts only and for single language too -> ctsd ? The benefit would be that we would never hit potential characters limit anymore. Same goes for product key below.

Reply:
Hi @butenkor - this change covers the case when we have a product with at least three variants and then: 
 - we get a productDraft with only one variant - so we move the rest of variants to a new anonymized product
 - after that, we get another productDraft with variant from the anonymized product so we move other variants again to the second anonymized product

When anonymizing product, we were moving the rest of variants to a new anonymized product and always adding salt to slug and key.
Now we will also check if the slug and key which we want to anonymize contain already salt and in that case, we will replace it with the new one.

So the original slug eg.: `{ en: 'en-slug-oldsalt', ctsd: 'oldsalt' }` will become: `{ en: 'en-slug-newsalt', ctsd: 'newsalt' }` instead of `{ en: 'en-slug-oldsalt-newsalt', ctsd: 'newsalt' }`

> I wonder if for the sake of consistency we should save salts only and for single language too - ctsd ?
We can replace slug and key only with salt. But I think that we hit the max length limit mainly because we anonymized product multiple times so the slug got too long. Therefore I would try this fix first.

Reply:
Thanks for explanation. In replace function it does not look like you keep original slug which was 'en-slug' and replace it completely with newsalt (which does not contain original slug anymore), right?

Reply:
Ok, forget it. I missed the point that ctsd contains salt only.